flink_jobs/status_trends/src/main/java/argo/functions/calculations/CalcServiceEndpointFlipFlop.java

```diff
@@ -0,0 +1,69 @@
+package argo.functions.calculations;
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+import argo.utils.TimelineBuilder;
+import argo.pojos.TimelineTrends;
+import argo.profileparsers.OperationsParser;
+import argo.utils.Utils;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.TreeMap;
+import org.apache.flink.api.common.functions.GroupReduceFunction;
+import org.apache.flink.util.Collector;
+
+/**
+ *
+ * @author cthermolia
+ *
+ * CalcServiceEndpointFlipFlop, count status changes for each service endpoint
+ * group
+ */
+public class CalcServiceEndpointFlipFlop implements GroupReduceFunction< TimelineTrends, TimelineTrends> {
+
+    private HashMap<String, String> opTruthTable;
+    private OperationsParser operationsParser;
+
+    public CalcServiceEndpointFlipFlop(OperationsParser operationsParser, String operation) {
+        this.operationsParser = this.operationsParser;
```

Comment:
I detect that this code is problematic. According to the [Dodgy code (STYLE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dodgy-code-style), [SA: Self assignment of local variable (SA_LOCAL_SELF_ASSIGNMENT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-assignment-of-local-variable-sa-local-self-assignment).
This method contains a self assignment of a local variable; e.g.
Such assignments are useless, and may indicate a logic error or typo.